e2e/test/iothub/device/MessageSendFaultInjectionTests.cs

```diff
@@ -237,46 +236,28 @@ public async Task Message_ThrottledConnectionLongTimeNoRecovery_AmqpWs()
         [DoNotParallelize]
         public async Task Message_QuotaExceededRecovery_Amqp()
         {
-            // act
-            Func<Task> act = async () =>
-            {
-                await SendMessageRecoveryAsync(
-                        new IotHubClientAmqpSettings(),
-                        FaultInjectionConstants.FaultType_QuotaExceeded,
-                        FaultInjectionConstants.FaultCloseReason_Boom)
+            await SendMessageRecoveryAsync(
+                    new IotHubClientAmqpSettings(),
+                    FaultInjectionConstants.FaultType_QuotaExceeded,
+                    FaultInjectionConstants.FaultCloseReason_Boom)
                 .ConfigureAwait(false);
-            };
-
-            // assert
-            var error = await act.Should().ThrowAsync<IotHubClientException>();
-            error.And.ErrorCode.Should().Be(IotHubClientErrorCode.DeviceMaximumQueueDepthExceeded);
-            error.And.IsTransient.Should().BeFalse();
```

Comment:
As `QuotaExceeded` is transient, the "SendMessageRecoveryAsync" call will succeed with retries. We won't expect `IotHubClientException` thrown out here.

Reply:
Should we mark quota exceeded as non-transient? Until the quota thresholds are refreshed retrying this operation won't help.

Reply:
This is a good question. I notice we are currently marking quota exceeded as transient for both hub device and service clients. I forgot why we are doing so, but let me check within our team if any objection of changing it to non-transient.

Reply:
It should be transient. Also remember that a retry policy can give up at some point (still retryable) and someone may have NoRetryPolicy in use.

Reply:
All quota I know of are time-based, meaning at some point the operation can be successful. It might be < a minute, or could be until the next day.

Reply:
I see, I believe this also explained why retrying for quota exceed error succeeded. I will leave it as transient then.